tag_release.py

```diff
@@ -0,0 +1,554 @@
+                    repo_name,
+                    branch_name,
+                    session=session,
+                )
+            except RequestException, ValueError:
+                msg = "Invalid branch {branch} in repo {repo}".format(
+                    branch=branch_name,
+                    repo=repo_name,
+                )
+                print(msg, file=sys.stderr)
+            continue
+        # are we specifying a parent repo?
+        parent_repo_name = repo_data["openedx-release"].get("parent-repo")
+        if parent_repo_name:
+            # we need the branch for the parent repo
+            parent_branch_name = repos[parent_repo_name]["openedx-release"]["branch"]
+            try:
+                to_tag[repo_name] = get_latest_commit_for_parent_repo(
+                    repo_name,
+                    parent_repo_name,
+                    parent_branch_name,
+                    session=session,
+                )
+            except RequestException, ValueError:
+                msg = "Problem getting parent ref for repo {repo}".format(
+                    repo=repo_name,
+                )
+                print(msg, file=sys.stderr)
+            continue
+    return to_tag
+
+
+def get_latest_commit_for_ref(repo_name, ref, session):
+    # is it a branch?
+    branch_url = "https://api.github.com/repos/{repo}/branches/{branch}".format(
+        repo=repo_name,
+        branch=ref,
+    )
+    branch_resp = session.get(branch_url)
+    if branch_resp.ok:
+        branch = branch_resp.json()
+        commit = branch["commit"]["commit"]
+        return {
+            "ref": ref,
+            "ref_type": "branch",
+            "sha": branch["commit"]["sha"],
+            "message": commit["message"],
+            "author": commit["author"],
+            "committer": commit["committer"],
+        }
+
+    # is it a tag?
+    tag_url = "https://api.github.com/repos/{repo}/git/refs/tags/{tag}".format(
+        repo=repo_name,
+        tag=ref,
+    )
+    tag_resp = session.get(tag_url)
+    if tag_resp.ok:
+        tag = tag_resp.json()
+        # need to do a subsequent API call to get the tagged commit
+        commit_url = tag["object"]["url"]
+        commit_resp = session.get(commit_url)
+        if commit_resp.ok:
+            commit = commit_resp.json()
+            return {
+                "ref": ref,
+                "ref_type": "tag",
+                "sha": commit["sha"],
+                "message": commit["message"],
+                "author": commit["author"],
+                "committer": commit["committer"],
+            }
+
+    msg = "No commit for {ref} in {repo}".format(
+        ref=ref, repo=repo_name,
+    )
+    raise ValueError(msg)
+
+
+def get_latest_commit_for_parent_repo(
+        repo_name, parent_repo_name, parent_branch_name, session,
+    ):
+    if parent_repo_name == "edx/edx-platform":
+        requirements_file = "requirements/edx/github.txt"
```

Comment:
BTW: Clinton & Feanil's plan to move to real PyPI releases is going to screw this up....